marimo/_save/hash.py

```diff
@@ -604,8 +623,12 @@ def cache_attempt_from_hash(
         context=context,
         pin_modules=pin_modules,
         hash_type=hash_type,
+        scoped_refs=scoped_refs,
     )
 
+    if as_fn:
```

Comment:
Leave comment, that defs are local to the function, so should not save them

```
def fn()
    global x
```

Could be an issue, but is already an anti-pattern